bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java

```diff
@@ -997,25 +908,32 @@ private void handleFavorites(String message) {
             Favorite f = null;
             boolean isTypePlaylist = false;
             for (String part : messageParts) {
+                final String decoded = decode(part);
+                int colonPos = decoded.indexOf(":");
+                if (colonPos < 0) {
+                    continue;
+                }
+
+                final String key = decoded.substring(0, colonPos);
+                final String value = decoded.substring(colonPos);
```

Comment:
Shouldn't this be `colonPos + 1`?

Reply:
Oops, of course. Thanks, fixed.

Reply:
NB: I thought about factoring out this parsing to a separate method, but due the lack of a destructuring binding in Java (e.g. Kotlin: `val (key, value) = ...` this didn't really improve readability.

Reply:
Ended up going a step back and parsed the full CLI response into a map.